### ISSUE TITLE
Setup jenkins build server + Test Kubernetes cluster

### ISSUE THREAD

**Issue by maclof** [maclof/kubernetes-client#11]
We need to improve test coverage as per issue #2 however in order to write functional tests we need a real test cluster to run them against.

**Comment by HD-CMS** [comment]
Hey @maclof,

maybe use a Travis.ci-Job for this: 

Base Image:
https://github.com/Mirantis/kubeadm-dind-cluster

Setup Cluster:
https://gist.github.com/cedriclam/7f667e28fdb4a710befa43434c8c9ab4

But then you need two `languages` (php, go) to test the complete code. I think currently this is not possible.

**Comment by maclof** [comment]
Good idea, but to be honest I don't mind hosting the test cluster myself, but I need to be able to inject the credentials into Travis CI without them being made public, not sure if that's possible.

**Comment by meme** [comment]
As a note: it's probably best to do this with GitHub Actions now. Adding secrets is trivial, and would enable this functionality testing.

**Comment by maclof** [comment]
GitHub Actions now runs the test matrix, and the repository includes scripts for provisioning and exercising a disposable single-node Kubernetes cluster. Closing as implemented.
